src/compiler/binder.rs

```diff
@@ -420,12 +421,8 @@ impl BinderType {
         name: __String,
     ) -> Rc<Symbol> {
         let symbol = self.create_symbol(symbol_flags, name).wrap();
-        match &*self.file() {
-            Node::SourceFile(source_file) => {
-                source_file.keep_strong_reference_to_symbol(symbol.clone());
-            }
-            _ => panic!("Expected SourceFile"),
-        }
+        enum_unwrapped!(&*self.file(), [Node, SourceFile])
```

Comment:
The next step of convenience that's tempting is to give the macros knowledge of the `Node`/`Type`/etc enum variants so that I could just write eg:
```
node_unwrapped!(&*self.file(), SourceFile)
```